lib/api.ts

```diff
@@ -1162,16 +1176,30 @@ module.exports = function (PersistObjectTemplate, baseClassForPersist) {
             });
     }
 
-    PersistObjectTemplate.beginTransaction = function () {
-        var txn = {
-            id: new Date().getTime(), dirtyObjects: {},
-            savedObjects: {}, touchObjects: {}, deletedObjects: {}, deleteQueries: {}
+    PersistObjectTemplate.beginTransaction = function (): PersistorTransaction {
+        return {
+            id: new Date().getTime(),
```

Comment:
Is this to guarantee uniqueness? any pros/cons over using UUID's?

Reply:
I believe the transaction id here is not functionally used - it's more for tracking a single transaction through the logs. so in this case I think either would be fine, not a huge benefit in either direction so I'll keep as is is for now. if we decided that we wanted to take more time to refactor / change anything in persistor related to unique ID's, we'd probably prioritize getting rid of mongo-bluebird library for object id generation...

Reply:
Makes sense to me!